src/Context/DataContext/DataProvider.js

```diff
@@ -0,0 +1,29 @@
+import { DataContext } from "./DataContext";
+import { useState, useReducer } from "react";
+import { reducerFunction } from "./ReducerFunction";
+
+export const DataProvider = ({ children }) => {
+  const [toast, setToast] = useState(false);
+  const [toastMessage, setToastMessage] = useState("");
+
+  const [state, dispatch] = useReducer(reducerFunction, {
+    products: [],
+    wishProducts: [],
+    cartProducts: [],
+  });
+
+  return (
+    <DataContext.Provider
+      value={{
+        toast,
+        setToast,
+        toastMessage,
```

Comment:
since you've added toast related states in context, toast for one for page will be visible in another page as well, if you visit quickly. So, try to fix it.